File: querydsl-core/src/test/java/com/mysema/util/ReflectionUtilsTest.java
package com.mysema.util;

import static org.junit.Assert.assertEquals;

import java.io.Serializable;
import java.util.Arrays;
import java.util.Collection;
import java.util.HashSet;
import java.util.List;

import org.junit.Test;


public class ReflectionUtilsTest {
    
    @Test
    public void getImplementedInterfaces(){
        assertEquals(
            new HashSet<Class<?>>(Arrays.<Class<?>>asList(Appendable.class, CharSequence.class, Serializable.class)), 
            ReflectionUtils.getImplementedInterfaces(StringBuilder.class));
        
        assertEquals(
            new HashSet<Class<?>>(Arrays.<Class<?>>asList(Collection.class, Iterable.class)), 
            ReflectionUtils.getImplementedInterfaces(List.class));
    }

}

File: querydsl-scala/src/test/scala/com/mysema/scala/ReflectionUtilsTest.scala
package com.mysema.scala

import java.lang.reflect._
import javax.annotation.Nullable
import scala.collection.mutable.ListBuffer

import org.junit._
import org.junit.Assert._

class ReflectionUtilsTest {

  @Test
  def getSuperClasses_of_String {
    assertEquals(List(classOf[String],classOf[AnyRef]), ReflectionUtils.getSuperClasses(classOf[String]))
  }
  
  @Test
  def getImplementedInterfaces {
    assertEquals(Set(classOf[java.io.Serializable],classOf[Comparable[_]],classOf[CharSequence]), 
        ReflectionUtils.getImplementedInterfaces(classOf[String]))
  }
  
  @Test
  def getFields {
    assertTrue( ReflectionUtils.getFields(classOf[String]).size > 0)
  }
  
  
  
}